lib/cc/engine/analyzers/violation.rb

```diff
@@ -86,7 +86,7 @@ def fingerprint
           digest << "-"
           digest << current_sexp.mass.to_s
           digest << "-"
-          digest << occurrences.to_s
+          digest << check_name
```

Comment:
Classic also uses the type of the sexp's root node, I think. That's an extra piece of info to make the fingerprint more unique that may be worth adding.